### ISSUE TITLE
DeviceAttribute constructor fails for short datatype (Regression from Tango 8)

### ISSUE THREAD

**Issue by t-b** [tango-controls/cppTango#392]
The following code 

```cpp
    std::string attr = "myname";
    short data = -111;
    Tango::DeviceAttribute da1;
    da1.set_name(attr);
    da1 << data;
    std::cout << "type: " << da1.get_type() << std::endl;

    Tango::DeviceAttribute da2(attr, data);
    std::cout << "type: " << da2.get_type() << std::endl;
```
gives with tango 9.2.5a

```
type: 100
type: 914331536
```

So the the constructor with signed short does not work at all, but the first method gives an unknown type. (DATA_TYPE_UNKNOWN is 100).

**Comment by bourtemb** [comment]
Good catch, Thomas!
Thanks for the bug report.
It seems this bug was introduced when DevEnum support was added in Tango 9.
It looks like DeviceAttribute `data_type` field is not initialized in this case.
The initialization is now necessary for the get_type() method to work as expected.

**Comment by Ingvord** [comment]
Resolved in 3721f6e

**Comment by t-b** [comment]
@bourtemb @Ingvord Thanks.
